### DISCUSSION TITLE
EdgeX Go services failed to start with error: "couldn't create database client: Unsupported database type"

### DISCUSSION THREAD

**Discussion by vediyappan-hcl** [edgexfoundry/community#133] [Q&A] [answered]
### Summary
Cloned the latest code from https://github.com/edgexfoundry/edgex-go, build it and tried to run the microservices; the services are failing to run. I have disabled the the edgex security by setting EDGEX_SECURITY_SECRET_STORE=false. Why does it requires to create database client? I simply want to run the latest microservices to test one of the device service with latest changes. Any help would be highly appreciated.

### What I did?
1. git clone https://github.com/edgexfoundry/edgex-go.git
2. cd edgex-go/
3. make clean
4. make build
5. Added  a shell script (named edgex-launch.sh) to run all the services together and kept in cmd/
6. Run the script

### What is expected?
All the microservices should start running.

### What is happening?
Microservices such as core-data, core-metadata, core-command, support-scheduler and support-notifications are failing to start with error: "couldn't create database client: Unsupported database type"

### EdgeX Logs:
_vediyappan@vediyappan-desktop:~/work/uart-upgrade/edgex-go/cmd$ ./edgex-launch.sh
/home/vediyappan/work/uart-upgrade
vediyappan@vediyappan-desktop:~/work/uart-upgrade/edgex-go/cmd$ level=INFO ts=2023-06-23T06:15:36.863320576Z app=core-metadata source=config.go:640 msg="Loading configuration file from res/configuration.yaml"
level=INFO ts=2023-06-23T06:15:36.864637046Z app=core-metadata source=config.go:226 msg="Private configuration loaded from file with 0 overrides applied"
level=INFO ts=2023-06-23T06:15:36.867146987Z app=core-metadata source=bootstrap.go:56 msg="Loaded unit of measure configuration from ./res/uom.yaml"
**level=WARN ts=2023-06-23T06:15:36.868046535Z app=core-metadata source=database.go:106 msg="couldn't create database client: Unsupported database type"**
level=INFO ts=2023-06-23T06:15:36.868904472Z app=core-command source=config.go:640 msg="Loading configuration file from res/configuration.yaml"
level=INFO ts=2023-06-23T06:15:36.871944353Z app=core-command source=config.go:226 msg="Private configuration loaded from file with 0 overrides applied"
level=INFO ts=2023-06-23T06:15:36.874432757Z app=core-command source=clients.go:164 msg="Using REST for 'core-metadata' clients @ http://localhost:59881"
**level=ERROR ts=2023-06-23T06:15:36.875066178Z app=core-command source=messaging.go:45 msg="MessageBus configuration not set or missing from service's GetBootstrap() implementation"**
level=INFO ts=2023-06-23T06:15:36.879591361Z app=core-data source=config.go:640 msg="Loading configuration file from res/configuration.yaml"
level=INFO ts=2023-06-23T06:15:36.881088681Z app=core-data source=config.go:226 msg="Private configuration loaded from file with 0 overrides applied"
**level=WARN ts=2023-06-23T06:15:36.882811499Z app=core-data source=database.go:106 msg="couldn't create database client: Unsupported database type"**
level=INFO ts=2023-06-23T06:15:36.884419726Z app=support-scheduler source=config.go:640 msg="Loading configuration file from res/configuration.yaml"
level=INFO ts=2023-06-23T06:15:36.890262361Z app=support-notifications source=config.go:640 msg="Loading configuration file from res/configuration.yaml"
level=INFO ts=2023-06-23T06:15:36.891996383Z app=support-notifications source=config.go:226 msg="Private configuration loaded from file with 0 overrides applied"
level=INFO ts=2023-06-23T06:15:36.892518267Z app=support-scheduler source=config.go:226 msg="Private configuration loaded from file with 0 overrides applied"
**level=WARN ts=2023-06-23T06:15:36.893265168Z app=support-notifications source=database.go:106 msg="couldn't create database client: Unsupported database type"**
**level=WARN ts=2023-06-23T06:15:36.893836348Z app=support-scheduler source=database.go:106 msg="couldn't create database client: Unsupported database type"**
level=WARN ts=2023-06-23T06:15:37.871350684Z app=core-metadata source=database.go:106 msg="couldn't create database client: Unsupported database type"
level=WARN ts=2023-06-23T06:15:37.88341695Z app=core-data source=database.go:106 msg="couldn't create database client: Unsupported database type"
level=WARN ts=2023-06-23T06:15:37.893620917Z app=support-notifications source=database.go:106 msg="couldn't create database client: Unsupported database type"
level=WARN ts=2023-06-23T06:15:37.894120802Z app=support-scheduler source=database.go:106 msg="couldn't create database client: Unsupported database type"_

### What does my script file contains?
```
This script was written manually to run natively built edgex microservices
#!/bin/usr/env bash

EDGEX_PATH="/home/vediyappan/work/uart-upgrade"

echo $EDGEX_PATH

cd $EDGEX_PATH/edgex-go/cmd/core-command
export EDGEX_SECURITY_SECRET_STORE=false
./core-command &

cd $EDGEX_PATH/edgex-go/cmd/core-metadata
export EDGEX_SECURITY_SECRET_STORE=false
./core-metadata &

cd $EDGEX_PATH/edgex-go/cmd/core-data
export EDGEX_SECURITY_SECRET_STORE=false
./core-data &

cd $EDGEX_PATH/edgex-go/cmd/support-scheduler
export EDGEX_SECURITY_SECRET_STORE=false
./support-scheduler &

cd $EDGEX_PATH/edgex-go/cmd/support-notifications
export EDGEX_SECURITY_SECRET_STORE=false
./support-notifications &
```

**Comment by bnevis-i** [comment, marked as answer]
Does https://github.com/orgs/edgexfoundry/discussions/131 provide insight to your problem?

**Reply by vediyappan-hcl** [reply]
@bnevis-i Yes, it helped in resolving the issue. Thanks!
